### PR TITLE
CONTRIBUTING.md: add sections on contributing code/documentation

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? 👉 **N/A**
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This adds brief instructions on contributing code and improving command documentation/help. The latter section was inspired by https://github.com/Homebrew/brew/pull/270#issuecomment-221330769. Minor adjustsments are also made to the styling/wording of the only section that looks different from what we usually use in this file and its [`homebrew/core` counterpart](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,23 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
...
+* create a pull request in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md)
+
+### Contribute code
+
+* there are no universally applicable rules (a lot depend on the scope of your change)
+* before you start coding, consider [proposing a feature](#propose-a-feature) (see above) or discussing the idea in an issue
+* to learn how to structure your commits, consult the Git history of modified files
+* [open a pull request](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) on this repository
+
+### Improve help for `brew <command>`
```
Comment (MikeMcQuaid) [inline .github/CONTRIBUTING.md]: I think this section is too specific for here. The longer we make this document the less likely people will read any of it. Would be good to pull into a separate "Contributing documentation changes" document that's linked from here.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -10,6 +10,23 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
 
 ### Propose a feature
 
-* Please create a pull request with your change proposal in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md).
+* create a pull request in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md)
+
+### Contribute code
+
+* there are no universally applicable rules (a lot depend on the scope of your change)
```
Comment (MikeMcQuaid) [inline .github/CONTRIBUTING.md]: Think this line is worth killing, it doesn't say much.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -10,6 +10,23 @@ First time contributing to Homebrew? Read our [Code of Conduct](https://github.c
 
 ### Propose a feature
 
-* Please create a pull request with your change proposal in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md).
+* create a pull request in the [Homebrew Evolution](https://github.com/Homebrew/brew-evolution) repository using the [proposal template](https://github.com/Homebrew/brew-evolution/blob/master/proposal_template.md)
+
+### Contribute code
```
Comment (MikeMcQuaid) [inline .github/CONTRIBUTING.md]: I'm not convinced we need this section, really. It's too generic to be particularly useful unless it's just a single bullet pointing to https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (UniqMartin) [inline .github/CONTRIBUTING.md]: Agreed. I just wanted to get this documented so it's easier for people to discover (or get linked to). Wasn't sure where to best put this, but a separate document seems reasonable.

**Comment by MikeMcQuaid** [comment]
@UniqMartin Can you address the comments and get this 🚢d? Thanks!

**Comment by UniqMartin** [comment]
> Can you address the comments and get this 🚢d? Thanks!

I will as soon as possible. Sorry for letting this linger for way too long …

**Comment by MikeMcQuaid** [comment]
Gentle ping on this.

**Comment by MikeMcQuaid** [comment]
Passing on this for now but will review another PR with changes made. Thanks!

**Pull request closed without merging**